apps/launch/app.js

```diff
@@ -81,8 +84,14 @@
         // cleanup the timeout to not leave anything behind after being removed from ram
         if (lockTimeout) clearTimeout(lockTimeout);
         Bangle.removeListener("lock", lockHandler);
+        // Restore widgets if they were hidden by fullscreen setting
+        if (global.WIDGETS) require("widget_utils").show();
```

Comment:
Do we also need:
```suggestion
        if (global.WIDGETS || settings.fullscreen) require("widget_utils").show();
```

or perhaps:
```suggestion
        if (hiddenWidgets) require("widget_utils").show();
```
(with that bool set when we call widget utils' `hide()`)

Reply:
I *think* we might be ok: https://github.com/espruino/BangleApps/blob/master/modules/widget_utils.js#L21

But it feels like a good idea to check anyway as otherwise it may force a redraw? Did you mean `if (global.WIDGETS && settings.fullscreen)` instead?

Reply:
Ah no - we're looking at it from different sides. Your side is whether it's safe to call (so I can see why you're thinking `&&` and the `widget._draw` check), mine is if we need to undo our changes, which are either if we've done a `.hide()` from having widgets:

https://github.com/espruino/BangleApps/blob/5dd73bbea83c535568a179d56b7478b7d489885c/apps/launch/app.js#L27-L28

or if we've had the fullscreen setting:

https://github.com/espruino/BangleApps/blob/5dd73bbea83c535568a179d56b7478b7d489885c/apps/launch/app.js#L91-L92

If either of those are true, we need to `.show()`, hence `global.WIDGETS || settings.fullscreen`, right?

Reply:
But if there are no widgets, there's no need to update their properties with the `.show()` call - right? There's nothing for `widget-utils` to act on.

Reply:
Yes, but then [the for loop](https://github.com/espruino/BangleApps/blob/ebbe4206e604660571020d712e4abea4d01c6e1a/modules/widget_utils.js#L19-L20) wouldn't be entered in that case so we're ok, still. And we're balancing our [above call to `.hide()`](https://github.com/espruino/BangleApps/blob/5dd73bbea83c535568a179d56b7478b7d489885c/apps/launch/app.js#L27-L28), which I think is as far as this code needs to go, and it can then trust that `widget_utils` will do the right thing

Reply:
Ahh, right - I see what you mean - but actually we only call `require("widget_utils").hide();` when we're supposed to be in fullscreen mode (line 27 is actually `if (!fullscreen) {..} else if (...WIDGETS) ...`. 

If we're not in fullscreen mode we almost certainly do have widgets but they're not hidden.

Reply:
Ah I hadn't spotted the fullscreen check, yeah I'm with you